### PR TITLE
circuit: some simplifications

### PR DESCRIPTION
**Pull request by tcoratger**
## Summary
What does this PR change?

Fixes #<issue-id>

## Why
What problem does this solve? Why is this approach correct?

## Changes
- [ ] Key change 1
- [ ] Key change 2

## Acceptance criteria
- [ ] <explicit condition that proves "done">
- [ ] <explicit condition that proves "done">

## Tests
- [ ] Unit tests added/updated
- [ ] Existing tests pass
- [ ] Edge cases considered (briefly note any)

## Benchmarks / Performance (if relevant)
- [ ] Before/after numbers included
- [ ] Benchmark command or link to CI perf output
- [ ] No unexpected regressions in CPU / memory / proof size / compile time

## Docs (if relevant)
- [ ] Rustdoc updated
- [ ] Book/README updated
- [ ] Examples updated

## Risk
What could break?

## Review notes
Where should reviewers focus on? Any tricky parts / follow-ups?

### CODE REVIEW

File: circuit/src/builder/circuit_builder.rs [hunk 1]
```diff
@@ -277,28 +277,29 @@ where
...
-                    .try_into()
-                    .expect("base_input length must equal 8"),
-            );
-            let mut output = Vec::with_capacity(width_ext);
-            for i in 0..width_ext {
-                let coeffs = &base_output[i * d..(i + 1) * d];
-                output.push(
-                    F::from_basis_coefficients_slice(coeffs)
-                        .expect("basis coefficients should be valid"),
+        let exec: crate::ops::poseidon2_perm::Poseidon2PermExec<F> =
```
Comment (Nashtare) [inline circuit/src/builder/circuit_builder.rs]: same here

```suggestion
        let exec: Poseidon2PermExec<F> =
```

File: circuit/src/ops/poseidon2_perm/plugin.rs [hunk 2]
```diff
@@ -77,37 +69,22 @@ fn lower_expr_slots(
 pub struct Poseidon2CircuitPlugin<F: Field> {
     type_id: NpoTypeId,
     poseidon2_config: Poseidon2Config,
-    config_payload: Poseidon2PluginConfig<F>,
+    npo_config: crate::op::NpoConfig,
```
Comment (Nashtare) [inline circuit/src/ops/poseidon2_perm/plugin.rs]:
```suggestion
    npo_config: NpoConfig,
```

File: circuit/src/ops/poseidon2_perm/plugin.rs [hunk 3]
```diff
@@ -77,37 +69,22 @@ fn lower_expr_slots(
...
 impl<F: Field> Poseidon2CircuitPlugin<F> {
-    pub fn new_ext(
+    pub fn new(
         config: Poseidon2Config,
         exec: Poseidon2PermExec<F>,
         trace_gen: TraceGeneratorFn<F>,
     ) -> Self {
         let type_id = NpoTypeId::poseidon2_perm(config);
-        let payload = Poseidon2PluginConfig::Ext(Poseidon2PermConfigData { config, exec });
+        let npo_config = crate::op::NpoConfig::new(Poseidon2PermConfigData { config, exec });
```
Comment (Nashtare) [inline circuit/src/ops/poseidon2_perm/plugin.rs]:
```suggestion
        let npo_config = NpoConfig::new(Poseidon2PermConfigData { config, exec });
```

**Review by Nashtare — COMMENTED** [review]
You've deleted entirely the Policy logic but this isn't reflected in README / mdbook.

**Comment by Nashtare** [comment]
> @Nashtare I think that the PR labeler thing still fails here probably because the PR was created before I entered the org, now this should be fixed for upcoming PRs

I do hope so, this is starting to bother me :D

**Comment by tcoratger** [comment]
@Nashtare Now I think all the conflicts are fixed
